tests/unit/store/actions.spec.ts

```diff
@@ -140,6 +140,16 @@ describe('actions', () => {
       ])
       done()
     })
+    it('checks if the subsection is still selected before committing the change', async (done) => {
```

Comment:
i do not understand how this test tests the assumption

Reply:
The tree selection changes between calling the action and resolving the promise returned by the mocked getter.
The action function notices this and does not commit the retrieved data.

Reply:
I've updated the test description to make it clearer